### DISCUSSION TITLE
link: support for forcing tracefs in kprobe/kretprobe

### DISCUSSION THREAD

**Discussion by brycekahle** [cilium/ebpf#941] [Q&A]
For the reasons specified in #880, would you be open to an option in `link.KprobeOptions` to force usage of `tracefs` and bypassing PMU completely?

**Comment by ti-mo** [comment]
Hey @brycekahle! I wouldn't be opposed to it, we already do this when the caller specifies maxactive.

Is having `kprobe_profile` the main reason, or do you have additional pain points?

**Reply by brycekahle** [reply]
`kprobe_profile` is the main reason.
